microservices/kubeApi/clients/ocp_routes.py

```diff
@@ -174,16 +180,35 @@ def prepare_apply_routes(ns, select_tag, hosts, root_path, data_plane, ns_templa
 
             route_template = ROUTES[templ_version]["ROUTE"]
 
-            # If host transformation is disabled, then select the appropriate
-            # SSL cert based on the suffix mapping
+            # Modified certificate selection logic
             ssl_ref = "tls"
-            if not settings.host_transformation['enabled']:
+            custom_cert_label= ''
+            custom_cert_found = False
+            
+            if is_host_custom_domain(host):
```

Comment:
Rather than duplicating the `is_host_custom_domain` function, I would just skip it, and assume that if the `certificates` is passed, then it is only including custom certs.. so just need to move the `logger.debug` line under the `if certificates`.  You can then delete the `is_host_custom_domain` for kube.

Reply:
I believe it's necessary to keep is_host_custom_domain - otherwise how to have kube raise an exception when a custom domain is used but no certs are provided? 

e.g. this test: https://github.com/bcgov/gwa-api/blob/feature/custom-certs/microservices/kubeApi/tests/clients/test_ocp_routes.py#L161

Reply:
kube can still raise an exception - if no cert is found an exception should be thrown